### ISSUE TITLE
Build failure - Ubuntu 20.04 Qt 5.15.2

### ISSUE THREAD

**Issue by nyue** [vsg-dev/vsgQt#11]
Is there a specific combination of versions that will work with Qt ?

I am build from the head of vsgQt

```
[4/9] Building CXX object src/vsgQt/CMakeFiles/vsgQt.dir/Debug/ViewerWindow.cpp.o
FAILED: src/vsgQt/CMakeFiles/vsgQt.dir/Debug/ViewerWindow.cpp.o 
/usr/bin/c++ -DQT_CORE_LIB -DQT_GUI_LIB -DQT_WIDGETS_LIB -DVK_USE_PLATFORM_XCB_KHR -DCMAKE_INTDIR=\"Debug\" -I/home/nyue/projects/vulkan/vsgQt/build/src/vsgQt/vsgQt_autogen/include_Debug -I/home/nyue/projects/vulkan/vsgQt/vsgQt_git/include -I/home/nyue/projects/vulkan/vsgQt/build/include -isystem /home/nyue/systems/qt/5.15.2/gcc/include -isystem /home/nyue/systems/qt/5.15.2/gcc/include/QtWidgets -isystem /home/nyue/systems/qt/5.15.2/gcc/include/QtGui -isystem /home/nyue/systems/qt/5.15.2/gcc/include/QtCore -isystem /home/nyue/systems/qt/5.15.2/gcc/./mkspecs/linux-g++ -isystem /home/nyue/systems/vsg/head/include -isystem /home/nyue/systems/vsgXchange/head/include -g -Wall -Wparentheses -Wno-long-long -Wno-import -Wreturn-type -Wmissing-braces -Wunknown-pragmas -Wmaybe-uninitialized -Wshadow -Wunused -Wno-misleading-indentation -Wextra -fPIC -std=gnu++17 -MD -MT src/vsgQt/CMakeFiles/vsgQt.dir/Debug/ViewerWindow.cpp.o -MF src/vsgQt/CMakeFiles/vsgQt.dir/Debug/ViewerWindow.cpp.o.d -o src/vsgQt/CMakeFiles/vsgQt.dir/Debug/ViewerWindow.cpp.o -c /home/nyue/projects/vulkan/vsgQt/vsgQt_git/src/vsgQt/ViewerWindow.cpp
/home/nyue/projects/vulkan/vsgQt/vsgQt_git/src/vsgQt/ViewerWindow.cpp: In member function ‘void vsgQt::ViewerWindow::intializeUsingAdapterWindow(uint32_t, uint32_t)’:
/home/nyue/projects/vulkan/vsgQt/vsgQt_git/src/vsgQt/ViewerWindow.cpp:171:31: error: ‘QVulkanInstance’ does not name a type; did you mean ‘vulkanInstance’?
  171 |     auto vulkanInstance = new QVulkanInstance;
      |                               ^~~~~~~~~~~~~~~
      |                               vulkanInstance
/home/nyue/projects/vulkan/vsgQt/vsgQt_git/src/vsgQt/ViewerWindow.cpp:177:9: error: ‘setVulkanInstance’ was not declared in this scope
  177 |         setVulkanInstance(vulkanInstance);
      |         ^~~~~~~~~~~~~~~~~
/home/nyue/projects/vulkan/vsgQt/vsgQt_git/src/vsgQt/ViewerWindow.cpp:179:45: error: ‘QVulkanInstance’ has not been declared
  179 |         auto surface = vsg::Surface::create(QVulkanInstance::surfaceForWindow(this), instance);
      |                                             ^~~~~~~~~~~~~~~
/home/nyue/projects/vulkan/vsgQt/vsgQt_git/src/vsgQt/ViewerWindow.cpp:187:16: error: type ‘<type error>’ argument given to ‘delete’, expected pointer
  187 |         delete vulkanInstance;
      |                ^~~~~~~~~~~~~~
ninja: build stopped: subcommand failed.
```

**Comment by vsg-dev** [comment]
There is a line in include/vsgQt/ViewerWindow.h to help detect the Qt version capabilities w.r.t Qt and choose the supported paths:

#define QT_HAS_VULKAN_SUPPORT (QT_VERSION >= QT_VERSION_CHECK(5, 10, 0))

So in theory it should work automatically.  Which version of Qt do you have installed?  Can you checked for the clas QVulkanInstance?

FYI, it's 4 months since I worked with vsgQt, at that point I was compiling on Ubuntu 21.10 and 18.04 and it worked in both cases.

**Comment by nyue** [comment]
OK. I will use 18.04 to learn vsgQt API/SDK for now.

When I become more familiar, I will see if I can fix Ubuntu 20.04 and eventually Ubuntu 22.04

**Comment by nyue** [comment]
I realised that the error I am getting on 20.04 was because I had a custom build of Qt 5.15.2

If I were to use the Qt that is shipped with Ubuntu 20.04, all is well.

**Comment by robertosfield** [comment]
Hi Nicholas,

On Mon, 21 Mar 2022 at 03:39, Nicholas Yue ***@***.***> wrote:

> I realised that the error I am getting on 20.04 was because I had a custom
> build of Qt 5.15.2
>
> If I were to use the Qt that is shipped with Ubuntu 20.04, all is well.
> Message ID: ***@***.***


Good to hear it's building with the Qt that comes with both 18.04 and
20.04.  I would like to get to the bottom of why there is problem with your
custom build of Qt 5.15.2.

What was the motivation behind the custom build?  Where there modification
to the build?

I'm trying to figure out why the  QT_HAS_VULKAN_SUPPORT macro definition
vsgQt uses doesn't correctly handle the features that Qt should be
providing - it's like your build of Qt 5.15.2 doesn't contain the Vulkan
functionality that it should do.
